app/src/test/java/pm/gnosis/heimdall/ui/onboarding/fingerprint/FingerprintSetupViewModelTest.kt

```diff
@@ -0,0 +1,56 @@
+package pm.gnosis.heimdall.ui.onboarding.fingerprint
+
+import io.reactivex.Observable
+import io.reactivex.observers.TestObserver
+import io.reactivex.subjects.PublishSubject
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.BDDMockito.given
+import org.mockito.Mock
+import org.mockito.junit.MockitoJUnitRunner
+import pm.gnosis.svalinn.security.EncryptionManager
+import pm.gnosis.tests.utils.ImmediateSchedulersRule
+
+@RunWith(MockitoJUnitRunner::class)
+class FingerprintSetupViewModelTest {
+    @JvmField
+    @Rule
+    val rule = ImmediateSchedulersRule()
+
+    private lateinit var viewModel: FingerprintSetupContract
+
+    @Mock
+    private lateinit var encryptionManagerMock: EncryptionManager
+
+    @Before
+    fun setup() {
+        viewModel = FingerprintSetupViewModel(encryptionManagerMock)
+    }
+
+    @Test
+    fun observeFingerprintForSetup() {
+        val testObserver = TestObserver.create<Boolean>()
+        val fingerprintSubject = PublishSubject.create<Boolean>()
+        given(encryptionManagerMock.observeFingerprintForSetup()).willReturn(fingerprintSubject)
+
+        viewModel.observeFingerprintForSetup().subscribe(testObserver)
+
+        fingerprintSubject.onNext(true)
+        testObserver.assertValueAt(0, true)
+        fingerprintSubject.onNext(false)
+        testObserver.assertValueAt(1, false)
```

Comment:
.assertValues(true, false)

then(encryptionManagerMock).should().observeFingerprintForSetup()
then(encryptionManagerMock).shouldHaveNoMoreInvocations()